File: notebooks/Row5_Antennapod_synchnull/app/src/main/java/com/example/row5antennapodsynchnull/MainActivity.java
package com.example.row5antennapodsynchnull;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.example.row5antennapodsynchnull.ui.main.ChaptersFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        Log.i("MainActivity", "onCreate");
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);

    }
    @Override
    protected void onResume(){
       super.onResume();
        Button b = findViewById(R.id.button);
        b.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                setContentView(R.layout.main_activity);
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.container, new ChaptersFragment())
                        .commitNow();
            }
        });
    }
    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.i("MainActivity", "onDestroy");
    }
}

File: testApps/kotlinCoroutines/KotlinExample/KotlinCoroutines/app/src/main/java/com/yashsoni/kotlincoroutines/MainActivity.kt
package com.yashsoni.kotlincoroutines

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.*
import okhttp3.OkHttpClient
import okhttp3.Request

class MainActivity : AppCompatActivity(), CoroutineScope by MainScope() {
    private var client = OkHttpClient()

    override fun onDestroy() {
        super.onDestroy()
        cancel()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn_click.setOnClickListener {
            launch {
                networkCallMain()
            }
        }
    }

    private suspend fun networkCallMain() {
        val result = networkCallHelper()
        withContext(Dispatchers.Main) {
            tv_result.text = result
        }
    }

    private suspend fun networkCallHelper(): String {
        return withContext(Dispatchers.Default) {
            val url = "http://demo2202897.mockable.io/testGet "
            val request = Request.Builder()
                .url(url)
                .build()

            val response = client.newCall(request).execute()
            return@withContext response.body()!!.string()
        }
    }
}